tests/src/CLRTest.Execute.Batch.targets

```diff
@@ -213,21 +217,83 @@ IF NOT "%CLRCustomTestLauncher%"=="" (
 )
       ]]></BatchCLRTestLaunchCmds>
       <BatchCLRTestLaunchCmds Condition=" '$(IlasmRoundTrip)'=='true' "><![CDATA[
-$(BatchCLRTestLaunchCmds)
-ECHO %LAUNCHER% $(TargetAssemblyName) %CLRTestExecutionArguments% %Host_Args%
-%LAUNCHER% $(TargetAssemblyName) %CLRTestExecutionArguments% %Host_Args%
-
-IF NOT "!ERRORLEVEL!"=="%CLRTestExpectedExitCode%" (
-  ECHO END EXECUTION OF IL{D}ASM BINARY - FAILED !ERRORLEVEL! vs %CLRTestExpectedExitCode%
-  ECHO FAILED
-  Exit /b 1
+  $(BatchCLRTestLaunchCmds)
+IF NOT DEFINED DoLink ( 
+  ECHO %LAUNCHER% $(TargetAssemblyName) %CLRTestExecutionArguments% %Host_Args%
+  %LAUNCHER% $(TargetAssemblyName) %CLRTestExecutionArguments% %Host_Args%
+
+  IF NOT "!ERRORLEVEL!"=="%CLRTestExpectedExitCode%" (
+    ECHO END EXECUTION OF IL{D}ASM BINARY - FAILED !ERRORLEVEL! vs %CLRTestExpectedExitCode%
+    ECHO FAILED
+    Exit /b 1
+  )
 )
       ]]></BatchCLRTestLaunchCmds>
       <BatchCLRTestLaunchCmds Condition="'$(CLRTestKind)' == 'BuildAndRun'"><![CDATA[
+
+set ExePath=$(InputAssemblyName)
+set LinkBin=__Link
+set Assemblies=-a System.Private.CoreLib
+set ReflectionRoots=
+
+IF defined DoLink ( 
+    IF NOT EXIST !ILLINK! (
```

Comment:
I've moved the linker testing and cleanup code to their own sections.
But unlinke ilasm/ildasm these sections will always be invoked.

The decision of whether to run via linker or not is not static, but done at runtime, since we don't want to rebuild the tests to tests with/without illink.